src/mesh/mod.rs

```diff
@@ -213,35 +226,72 @@ impl ShapeMesh {
             }
         }
 
-
         Ok(())
     }
 
-    /// Returns the point on the shape's surface the camera is currently looking at
-    pub fn get_focus(&self, camera: &Camera) -> Option<Point3<f32>> {
+    /// Returns equally distributed points on the near_plane. The number of points
+    /// returned is focus_points².
+    pub fn get_focii(&self, camera: &Camera, focus_points: u8) -> Vec<Point3<f32>> {
+        // sets the amount of focus points that may cause a split event
         const EPSILON: f32 = 0.000_001;
         const MAX_ITERS: u64 = 100;
 
-        let mut pos = camera.position;
-        for _ in 0..MAX_ITERS {
-            let distance = self.shape.min_distance_from(pos);
-            pos += camera.direction() * distance;
-            if distance < EPSILON {
-                break;
+        let bb = camera.get_near_plane_bb();
```

Comment:
It's probably easier to read if you say `let (top_left, bottom_right) = camera.get_near_plane_bb();`. It's always encouraged to destructure tuples instead of using `.0` and so on.